src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -3,45 +3,41 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { IncomingMessage } from 'http'
-import * as request from 'request'
 import { getLogger, Logger } from '../logger'
+import { getResponseFromGetRequest } from '../utilities/requestUtils'
 import { ResourceFetcher } from './resourcefetcher'
 
 export class HttpResourceFetcher implements ResourceFetcher {
     private readonly logger: Logger = getLogger()
 
-    public constructor(private readonly url: string) {}
+    public constructor(private readonly url: string, private readonly friendlyName?: string) {}
```

Comment:
This change makes all URLs opaque, which is not how some of the existing usage is intended. One example is when loading endpoints data: 

```
2020-10-27 08:36:19 [VERBOSE]: Loading resource from https://idetoolkits.amazonwebservices.com/endpoints.json
2020-10-27 08:36:20 [VERBOSE]: Finished loading resource from https://idetoolkits.amazonwebservices.com/endpoints.json
```

It should be possible for callers to use `HttpResourceFetcher` in a way that displays or masks the url, at the caller's discretion.

Reply:
I'm not sure how this makes it opaque? The URL is a constant and the key is a constant as well. Is there that much value in showing any of the URLs that I have masked?

Reply:
If I understand the change, this class will now never emit a url to logs. This means all existing usages where urls were intended to be written to logs are affected, including the example above. There are places where providing a URL makes it easier to understand/troubleshoot/diagnose what is happening, particularly if/when issues arise in the toolkit. With the endpoints example, this provides a way to understand where the Toolkit is connecting to, and try to reproduce any download failures outside the toolkit.

Reply:
So as of now, this change masks the following URLs:
* Endpoint XML
* Lambda sample request manifest
* Lambda getFunction URL (must mask)
* VSDBG install script
* SFN webview assets (JS/CSS)

I'm not convinced there's a lot of value in surfacing any of those URLs, again, since they're all constants and we can find the location easily. Is the concern that someone would have the URL blocked by a firewall rule that they'd have to override?

At the very least, surfacing the URL will have to be an opt-in. I'd rather make showing the URL a conscious choice to help short-circuit some of these issues in the future. Maybe have a mandatory `isUserData` param in the resource fetcher?

Reply:
* Endpoint XML, Lambda sample request manifest, VSDBG install script
These URLs are verbose level information that articulates what the toolkit is doing, and help to troubleshoot in relevant circumstances. They give users a chance to self-diagnose where applicable ("oh hey, I have that domain blocked"), and make it easier for us to assist when logs are shared in issues. They are constant values today, but could change for reasons we don't know today. Making them opaque would require knowledge of which version is running (which we might not have), in order to look up the corresponding value. Logging the URL avoids those additional lookup steps.

* Lambda getFunction URL / SFN webview assets (JS/CSS)
These are the URLs you're interested in masking with this PR

Some options are:
* Maintain the pre-PR behavior of `HttpResourceFetcher` as the default behavior (ie: not backwards incompatible). Add an optional configuration object to the constructor, which can mask the URL, and optionally provide a mask text.
* Leave HttpResourceFetcher as-is, and add a class that derives from it, which does URL masking.

Reply:
> *    Maintain the pre-PR behavior of HttpResourceFetcher as the default behavior (ie: not backwards incompatible). Add an optional configuration object to the constructor, which can mask the URL, and optionally provide a mask text.
> *    Leave HttpResourceFetcher as-is, and add a class that derives from it, which does URL masking.

I _really_ don't like the idea of having it show the URL by default; it should be opt-in so the dev has to think about whether or not it should be shown. The change itself would be minimal...it would be similar to the change I put here with the friendlyName param.

>Lambda getFunction URL / SFN webview assets (JS/CSS)
>These are the URLs you're interested in masking with this PR

Why would SFN fall into this category?

Reply:
> Why would SFN fall into this category?

Whoops, that's a mistake. I'll edit the comment